svelte.dev/packages

benmccann opened the pull request:
This is based off the work from @PuruVJ in https://github.com/sveltejs/svelte.dev/tree/registry-2

I copied over only the svelte.dev/packages homepage without the search functionality or full registry (only the subset of packages we're displaying).

jycouet commented on apps/svelte.dev/src/routes/packages/packages-meta.ts:
```diff
@@ -0,0 +1,312 @@
+} satisfies Record<string, { prompt: string }>;
+
+const SV_ADD = {
+	packages: [
+		'drizzle-orm',
+		'eslint',
+		'lucia',
+		'mdsvex',
+		'@inlang/paraglide-js',
+		'playwright',
+		'prettier',
+		'storybook',
+		'tailwindcss',
+		'vitest'
+	],
```
The list has 10 items, `sv` has 12 official add-ons, and there is 3 on the ui (lucia, mdsvex and paraglide)

Some add-ons has more than 1 dep, but maybe we can make sure that at least one is present ? (eg: eslint). Maybe it's ok to not have devtools-json also.

benmccann replied:
Thanks for catching this! I've added the missing ones

The other missing one is https://svelte.dev/docs/cli/sveltekit-adapter, but not quite sure how to display it in this format. I guess it could be `@sveltejs/adapter-auto`? Don't know if we really need it though because people would already expect that as an integration and we have a whole SvelteKit adapters category

jycouet commented on apps/svelte.dev/src/routes/packages/packages-meta.ts:
```diff
@@ -0,0 +1,312 @@
+import type { Package } from '$lib/server/content';
+
+const TAGS = {
+	'design-system': {
+		prompt:
+			'MUST contain 2+ reusable Svelte components designed as a cohesive set with shared design principles. Examples: complete UI kits, design systems, form libraries, widget collections, component libraries. Key indicator: components are intentionally built to work together, share styling patterns, and serve a unified purpose. NOT a single component or unrelated components bundled together.'
+	},
```
> The scripts for updating the registry have also not been copied over

Maybe `TAGS` is not needed?

benmccann replied:
thanks. removed for now. we can add it back if needed, but there was a lot of hesitancy around the AI portion, so I'm just leaving it out for now until/unless it's something agreed upon

jycouet reviewed (commented):
Looks nice 🎉

I can do the 2-3 comments, just let me know.

benmccann commented on apps/svelte.dev/src/routes/packages/packages-meta.ts:
```diff
@@ -0,0 +1,312 @@
+	alias: {
+		'@inlang/paraglide-js': 'paraglidejs',
+		'drizzle-orm': 'drizzle'
+	}
```
It looks like it's to make it match the adder name

jycouet replied:
We could do this:
<img width="810" height="266" alt="image" src="https://github.com/user-attachments/assets/b0f35ab2-0228-4fe2-bffd-19aa784230cd" />

Kind of nice things is that it can be also under like:
<img width="899" height="519" alt="image" src="https://github.com/user-attachments/assets/74969fbb-8ce5-4a62-999a-377c70199ce7" />

benmccann replied:
@Rich-Harris did most of the design and is pretty good at this stuff, so I'd be curious for his opinion on the addition of the Svelte icons and "npx sv add" commands and whether there might be a better way to accomplish this

teemingc commented:
Some thoughts from using it on mobile:
- comparing it with https://astro.build/integrations/ I think our card design is a lot sleeker. Nice job!
- I tried tapping the cards to see what would happen only to realise I had to click the icons on the bottom right to learn more. Maybe clicking the package title/desc should link to the package website?
- There are a lot of categories and it can be a little overwhelming on mobile to scroll up and down to view them all. Would a sub navigation list at the top of the page help with this? Or are we waiting for the filter and search?
- Seeing `sv add` as a newcomer can be confusing. Is it worth explaining what it is or linking to the docs in a description?

benmccann commented:
Having the card go to the website seems like a good idea
Agree "sv add" is a confusing title. At the very least it should be "sv add-ons" with that linked to the CLI docs
I think we should also have each card in the sv section go to the CLI docs for that particular add-on

benmccann commented:
> what are our criteria for inclusion?

Basically ambassadors and maintainers nominate things that are good or not and vote on them to add or remove things. We should definitely have everyone do a pass on this once we've got the rest settled. The content hasn't been thoroughly vetted. The router category hadn't been reviewed at all, so I just cleaned that up a bunch. Reasonable people will disagree on many of the others, but I think they're at least defensible even if not perfect. wuchale looks brand new and not very used, so not surprised it isn't listed here. paraglide even if only infrequently developed is still one of the better libraries out there. For sswr, remote functions only work for sveltekit and non-sveltekit users may still want options. superforms is still being worked on even if a bit behind.; From the superforms repo: 

> "With both the Runes update and remote functions, I'm leaning towards a full rewrite for v3"

teemingc commented:
There's a Windows Firefox issue with horizontal scrolling where it stops at each card instead of a smooth scroll to the end. I don't think we can do anything about this though.

https://github.com/user-attachments/assets/03c30777-f39f-4a52-b7f5-a53ddb5538dd

In contrast, Windows Chrome has a smooth scroll but it has a really janky snapping behaviour at the end. An alternative is to use `scroll-snap-type: proximity` instead of `mandatory`.

benmccann commented:
> I'm also wondering what's a good way to add a link to the sv CLI docs in the description. Can the description data contain anchor links or is it just text?

Yeah, I agree. This was something I had been wanting to do. I updated it so that it can contain links and added links in the various descriptions

> we could ditch the carousel. If that makes the page too tall, then we could put all items after the first n (4?) in a category behind a 'show more'

The thing I really like about the carousels is that it's pretty fast to be able to scroll to the bottom and see all the categories since each category doesn't take up much room. If we were to investigate a different design it'd be nice to keep the ability to quickly see all categories - maybe with a table of contents on the side or something

dominikg commented:
The horizontal scroll feels hard to use with a mouse on my linux desktop. There are not too many elements per category, can we just wrap them (with collapsibles if you want to show more categories per screen) or make sub-pages per category with all of them and just show 1 row worth on the listing page?

dominikg commented on .github/workflows/sync-packages.yml:
```diff
@@ -0,0 +1,38 @@
+        with:
+          node-version: 22
+          cache: pnpm
+      - run: pnpm install --frozen-lockfile
```
```suggestion
      - run: pnpm install --frozen-lockfile --ignore-scripts
```

don't run any scripts to make it less likely that something can corrupt the repo, esp with contents: write permission

jycouet commented on apps/svelte.dev/package.json:
```diff
@@ -13,7 +13,8 @@
 		"format": "prettier --write .",
 		"lint": "prettier --check .",
-		"sync-docs": "tsx scripts/sync-docs/index.ts"
+		"sync-docs": "tsx scripts/sync-docs/index.ts",
```
Going to look later 👍

benmccann replied:
I removed `tsx` for the new script. It's too involved to remove it for the existing script at the moment, so better to do it in a separate PR to avoid making this one too large

benmccann commented on apps/svelte.dev/src/lib/packages-meta.ts:
```diff
@@ -246,7 +246,8 @@ const FEATURED: {
 		title: 'More',
-		description: 'These are just a few highlights. See a larger <a href="https://www.sveltesociety.dev/packages">directory of packages at sveltesociety.dev</a>.',
+		description:
+			'These are just a few highlights. See a larger <a href="https://www.sveltesociety.dev/packages">directory of packages at sveltesociety.dev</a>.',
```
It looks like Kev might have accidentally disabled the updates when switching the v2 branch to `main`. I just updated it manually. Hopefully that can hold us over until he gets the v2 site launched.

Rich-Harris commented:
Reimplemented it using `<details>`. The one small detail that isn't great is that the `<summary>` element is always before the revealed items in the tab order, even though it's visually at the end of the section. We _could_ fix it when JS is available by using a `<button>` instead, though it's finicky so I'll leave that as a problem for future us.

Can we get rid of the `#readme` suffix on the repo URLs?
